pyresample/bucket/__init__.py

```diff
@@ -28,6 +29,39 @@
 LOG = logging.getLogger(__name__)
 
 
+@dask.delayed(pure=True)
+def _sort_weights(statistic_method, weights):
+    """Sort idxs and weights based on weights."""
+    if statistic_method == 'min':
+        order = np.argsort(weights)
+    elif statistic_method == 'max':
+        order = np.argsort(weights)[::-1]
+
+    return order
+
+
+@dask.delayed(pure=True)
+def _get_bin_statistic(bins, idxs_sorted, weights_sorted):
+    """Get the statistic of each bin."""
+    # get where the `idxs_sorted` located in `bins`
+    binned_values = np.digitize(idxs_sorted, bins, right=True)
+
+    # mask value outside of bin
+    binned_values_masked = np.ma.masked_array(binned_values,
+                                              (idxs_sorted < bins.min()) | (idxs_sorted > bins.max()))
+
+    # get the first index and value in each bin
+    unique_bin, unique_idx = np.unique(binned_values_masked, return_index=True)
```

Comment:
These could be grouped together in `_find_unique_bins_and_indices()` function, and I think the comments could be removed.

Reply:
Done in 1aa2779.